### PR TITLE
Secure connection

### LINKED ISSUE #18: ensure frontend redirects to login page if backend refuses requests
This is required s.t. our application is only available via the TU VPN.

The backend refuses requests if either not logged in (#25 ) or not inside TUDA network (#17 ), in this case a sensible error message should be displayed and the user should be redirected to the login page.

### LINKED ISSUE #25: password protection / authentication
As a VLA member, I want to access this application. I enter the URL into a browser and get a password prompt. After entering the password, I see the calendar view. Since there is only one password for all my colleagues in VLA, I can see the same exact information as all my colleagues do.

### PR DESCRIPTION
**Pull request by MarcoLeipnitz**
Initial draft for the authentication.

closes #18 
closes #25

---

### Implementation

- [x] Does this code change accomplish what it is supposed to do?
- [x] Can this solution be simplified without unreasonable effort
- [x] Is the code modular enough?
- [x] Does similar functionality already exist in the codebase? If yes, why isn’t it reused?
- [x] Single Responsibility Principle: Does any class serve only one purpose? If not, it should probably split up in multiple classes.
- [x] Interface Segregation: If interface implemented, is it fully implemented?
- [x] Dependency Injection: Are central object instantiation processes managed by a single context?
- [x] If any functionality is missing, is it marked with a comment that starts with TODO:?
- [x] Do you see any potential to improve the performance of the code significantly? 

### Logic Errors and Bugs

- [x] Can you think of any use case in which the code does not behave as intended?
- [ ] Can you think of any inputs or external events that could break the code?
- [ ] Are edge cases covered by tests?

### Error Handling and Logging

- [ ] Is error handling done the correct way?
- [ ] Should any logging or debugging information be added or removed?
- [x] Are error messages user-friendly?

### Dependencies (internal and external)

- [ ] Were updates to documentation, configuration, or readme files made as required by this change?
- [ ] Is the API.md up-to-date?
- [ ] Are the enumerations between frontend and backend synchronized?

### Security and Data Privacy

- [x] Does the code introduce any security vulnerabilities?
- [x] Is (user) input validated, sanitized, and escaped to prevent security attacks such as cross-site scripting or SQL injection?
- [x] Is sensitive data like user data handled and stored securely?
- [x] Is the right encryption used?
- [x] Is data retrieved from external APIs or libraries checked for security issues?
- [x] Were any secrets committed?

### Testing and Testability (later)

- [ ] Have automated tests been added, or have related ones been updated to cover the change?
- [ ] Do the existing tests reasonably cover the code change?
- [ ] Are there some test cases, input or edge cases that should be tested in addition?

### Usability and Accessibility

- [x] Is the proposed solution (UI) accessible (e.g. colorblindness, ...)?
- [x] Is the API/UI intuitive to use?
- [ ] Was the app tested on another device?

### Readability

- [x] Is the code easy to understand?
- [ ] Which parts were confusing to you and why?
- [ ] Can the readability of the code be improved by smaller methods/different names?
- [x] Is the code located in the right file/folder?
- [ ] Do you think certain methods should be restructured to have a more intuitive control flow?
- [ ] Are there redundant or outdated comments?
- [x] Could some comments convey the message better?
- [x] Could some comments be removed by making the code itself more readable?

### CODE REVIEW

**Comment by MarcoLeipnitz** [comment]
This branch will probably deal with 
#17 Backend answering only to TUDa IP addresses 
#18 Redirecting to login page if not authenticated
#25 Password protection and proper authentication

**Comment by MarcoLeipnitz** [comment]
We should discuss if id should really be the identifier rather than the name.
Logging in with an ID has already proven unnatural (for me at least)

**Comment by MarcoLeipnitz** [comment]
This is ready for review, just not ready to be merged as we have to discuss #17 with the VLA before that.

**Comment by MarcoLeipnitz** [comment]
When #89 is in main, we can start the merging process for this PR

File: docker-compose.services.yml [hunk 1]
```diff
@@ -42,7 +40,6 @@ services:
     build:
       context: .
       dockerfile: Dockerfile.prod
-    container_name: vlaorga_prod
```
Comment (JohaJung) [inline docker-compose.services.yml]: Oops. Actually, I didn't mean to commit that. But it is useful when testing around different branches, since otherwise docker complains about „container with that name already exists“.

File: scripts/getHashedPassword.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+if [ $# -ne 1 ]; then 
+    echo "Usage: $0 password";
+    exit 1;
+fi
+
+curl "http://localhost:8080/userManagement/unauthenticated/passwordConversion" \
+     -d "password=$1" || \
+(echo "The server must be running for this script to operate" && exit 1)
```
Comment (JohaJung) [inline scripts/getHashedPassword.sh]: newline at EOF

File: VLA-Backend/src/main/resources/static/login.html [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            .then(response => response.json())
+            .then(data => {
+                document.getElementById('csrf').value = data.token;
+            })
+            .catch(err => {
+                console.log("CSRF could not be fetched");
+                console.log(err);
+
+                // TODO: Integrate this properly
+                document.body.innerHTML +=
```
Comment (JohaJung) [inline VLA-Backend/src/main/resources/static/login.html]: Is this ToDo still up to date? What does it mean?

File: VLA-Backend/src/main/resources/application.properties [hunk 4]
```diff
@@ -1,10 +1,11 @@
 spring.application.name=VLA-Backend
 # Main datasource
+spring.profiles.default=prod
 spring.datasource.vla.url=${VLAORGA_DATASOURCE_URL}
 spring.datasource.vla.username=${VLAORGA_DATASOURCE_USERNAME}
 spring.datasource.vla.password=${VLAORGA_DATASOURCE_PASSWORD}
 spring.datasource.vla.driver-class-name=org.postgresql.Driver
-# Read-only datasource for the external linus application.
+# TODO: Change this before prod (DB schema handling)# Read-only datasource for the external linus application.
```
Comment (JohaJung) [inline VLA-Backend/src/main/resources/application.properties]: What should be changed before prod? -> move this comment to the right line, or formulate it more precisely.

**Review by JohaJung — CHANGES_REQUESTED** [review]
Some minor points, looks good otherwise.

**Review by JohaJung — APPROVED** [review]
